neuron_morphology/pipeline/post_data_to_s3.py

```diff
@@ -0,0 +1,101 @@
+import os
+import boto3
+import json
+from zipfile import ZipFile, ZIP_DEFLATED
+from io import BytesIO
+from argschema import ArgSchemaParser
+from neuron_morphology.pipeline._schemas import InputParameters
+
+def zip_files(file_list):
+    """
+    zip files into an archive in memory
+
+    Parameters
+    ---------------
+    file_list: file paths or file in bytes to be archived
+
+    Return
+    ---------------
+    archive: BytesIO obj
+
+    """
+
+    archive = BytesIO()
+
+    with ZipFile(archive, "a", ZIP_DEFLATED, False) as zip_file:
+        for name, file in file_list.items():
+            if isinstance(file, str):
+                with open(file, 'rb') as fh:
+                    data = BytesIO(fh.read())
+                zip_file.writestr(name, data.getvalue())
+            elif isinstance(file, BytesIO):
+                zip_file.writestr(name, file.getvalue())
+            else:
+                raise TypeError("Invalid input file!")
+
+    return archive
+
+
+def post_files_to_s3(archive_name, file_list, bucket_name):
+    """
+    This zip files to an archive in memory and post it to S3 bucket
+
+    Parameters
+    ------------------
+    archive_name: the archive's name in s3
+    file_list: file paths or files in bytes to be archived
+    bucket_name: s3 bucket name
+    
+    Return
+    ------------------
+    True if successful
+    
+    """
+
+    archive = zip_files(file_list)
+
+    aws_id = os.getenv('aws_access_key_id')
+    aws_key = os.getenv('aws_secret_access_key')
+    
+    s3_client = boto3.client('s3')
+
+    response = s3_client.put_object(Bucket=bucket_name, 
+                                Key=archive_name, 
+                                Body=archive.getvalue())
+
+
+def main():
+    """
+    Usage:
+    python post_data_to_s3.py --input_json INPUT_JSON
+    """
+    parser = ArgSchemaParser(schema_type=InputParameters)
+    inputs = parser.args
+
+    file_list = {}
```

Comment:
you call it a `file_list`, but it is actually a dict

Reply:
revised